src/tool/GenerateAction.cpp

```diff
@@ -74,12 +74,38 @@ DoGenerateAction()
     // Load the compilation database file
     //
     // --------------------------------------------------------------
-    MRDOCS_CHECK(toolArgs.inputPaths, "The compilation database path argument is missing");
-    MRDOCS_CHECK(toolArgs.inputPaths.size() == 1,
-        formatError(
-            "got {} input paths where 1 was expected",
-            toolArgs.inputPaths.size()));
-    auto compilationsPath = files::normalizePath(toolArgs.inputPaths.front());
+    if (detail::failed(toolArgs.inputPaths) && detail::failed(toolArgs.generateCompilationDatabase))
+    {
+        report::error("The compilation database path argument is missing or you can use --generate-compilation-database");
```

Comment:
I find this confusing. Even if `--generate-compilation-database`, mrdocs still needs to know the input path (which will now be the project path instead of compile_commands.json path).

I have the impression this is basically unrolling the `MRDOCS_CHECK` macros here (and forgetting to unroll one of them). This could be something like:

```
    MRDOCS_CHECK(!toolArgs.inputPaths.empty() || toolArgs.generateCompilationDatabase, "The input path argument is missing"); // <- not compilation database anymore
    MRDOCS_CHECK(toolArgs.inputPaths.size() == 1,
        formatError(
            "got {} input paths where 1 was expected",
            toolArgs.inputPaths.size()));
```